### PR TITLE
Create control select component

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

This component will be used for fan speed, alarm control panel mode, select options, etc...

CleanShot 2023-03-14 at 15 01 33

### Demo

https://deploy-preview-15819--home-assistant-gallery.netlify.app/#components/ha-control-select

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
This looks good! For a next iteration, maybe we should add a sliding animation when selecting a new value, and allow the user to slide the button? 😄

File: src/components/ha-control-select.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    this.addEventListener("keydown", this._handleKeydown);
+  }
+
+  destroyListeners() {
+    this.removeEventListener("focus", this._handleFocus);
+    this.removeEventListener("blur", this._handleBlur);
+    this.removeEventListener("keydown", this._handleKeydown);
+  }
+
+  _handleFocus() {
```
Comment (bramkragten) [inline src/components/ha-control-select.ts:85]:
```suggestion
  private _handleFocus() {
```

File: src/components/ha-control-select.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    super.connectedCallback();
+    this.setupListeners();
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.destroyListeners();
+  }
+
+  setupListeners() {
```
Comment (bramkragten) [inline src/components/ha-control-select.ts:73]:
```suggestion
  private _setupListeners() {
```

**Comment by piitaya** [comment]
> This looks good! For a next iteration, maybe we should add a sliding animation when selecting a new value, and allow the user to slide the button? 😄

Yes absolutely ! It can be nice (but in another PR)

**Review by bramkragten — APPROVED** [review]

**Pull request merged**